### PR TITLE
fix(curriculum): improve testing in regex challenge

### LINKED ISSUE #48225: Check For Mixed Grouping of Characters needs additional tests and different solution for testing
### Describe the Issue

The [Check For Mixed Grouping of Characters](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters) challenge really should have an extra test (or two) to make sure the regex tested does not allow the strings `FranklinRoosevelt` or `EleanorRoosevelt`to evaluate to `true` as would be with the actual solution (shown below) for the challenge.
```js
let myString = "Eleanor Roosevelt";
let myRegex = /(Franklin|Eleanor).*Roosevelt/;
let result = myRegex.test(myString);
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters

### Your code

n/a

### Expected behavior

The following tests should be added as such:
> Your regex myRegex should return `false` for the string `FranklinRoosevelt`

```js
myRegex.lastIndex = 0;
assert(!myRegex.test('FranklinRoosevelt'));
```

> Your regex myRegex should return `false` for the string `EleanorRoosevelt`

```js
myRegex.lastIndex = 0;
assert(!myRegex.test('EleanorRoosevelt'));
```

Also, a new solution (like below) should replace the existing one which would pass the new and all the existing tests.
```js
let myString = "Eleanor Roosevelt";
let myRegex = /(Franklin|Eleanor) ([A-Z]\.? )?Roosevelt/; // Change this line
let result = myRegex.test(myString); // Change this line
```
And for good measure, the following test could be added:

> Your regex myRegex should return `false` for the string `FranklinD. Roosevelt`

```js
myRegex.lastIndex = 0;
console.log(!myRegex.test('Franklin.D Roosevelt'));
```

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nayabatir1**
Checklist:

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48225 

I've updated the solution regex.

Provided solution `/(Franklin|Eleanor) ([A-Z]\.? )?Roosevelt/`
Eleanor asd. Roosevelt -> FAILS
Eleanor AFASDF. adfcx Roosevelt -> FAILS

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md [hunk 1]
```diff
@@ -85,6 +99,6 @@ let result = false; // Change this line
 
 ```js
 let myString = "Eleanor Roosevelt";
-let myRegex = /(Franklin|Eleanor).*Roosevelt/;
+let myRegex = /(Franklin|Eleanor)\s{1}(\S+\s)*Roosevelt/;
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md:102]: This regex would pass `Eleanor ?dsg Roosevelt` as valid.  That is a problem.

How about:
```
/(Franklin|Eleanor) (([A-Z]\.?|[A-Z][a-z]+) )?Roosevelt/
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my concern and suggestion.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md (hunk 1 above)
Comment (nayabatir1) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md:102]: Your regex is failing in this case
image

So I've modified my regex. Shall I use this ?
`(Franklin|Eleanor)\s([A-Za-z]+\.?\s)*Roosevelt`

**Comment by ieahleen** [comment]
Please take a moment and update the title of the PR so that it follows the [contributing guidelines](https://contribute.freecodecamp.org/)

**Comment by nayabatir1** [comment]
@ieahleen my pr includes two test cases and modified regex exp of the solution.
If you want to include what test cases I've added, the title is going to be way too long.

**Comment by naomi-lgbt** [comment]
An appropriate PR title could be something like `fix(curriculum): improve testing in regex challenge`.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md (hunk 1 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md]: For your new regex, `Eleanor dsg. asdf Roosevelt` should fail anyway as it is only supposed to allow for a middle name and not two and definitely not `dsg.`  A valid abbreviation would be a single capital letter followed by a period like 'D.`

**Review by naomi-lgbt — APPROVED** [review]
I think this should be good now?

**Review by moT01 — APPROVED** [review]
LGTM 🎉

What do you think @RandellDawson?

For future reference @nayabatir1, make sure to complete the checklist in the PR description please.

**Review by RandellDawson — APPROVED** [review]
I think these new changes have it covered better now.

**Comment by naomi-lgbt** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**